src/metabase/api/table.clj

```diff
@@ -551,4 +552,24 @@
                 :file     (get-in raw-params ["file" :tempfile])
                 :action   ::upload/replace}))
 
+(defn- delete-csv!
+  "This helper function exists to make testing the DELETE /api/table/:id endpoint easier."
+  [table-id]
+  (try
+    (let [table  (api/check-404 (t2/select-one :model/Table :id table-id))
+          result (upload/delete-upload! table)]
+      {:status 200
+       :body   (= :done result)})
+    (catch Throwable e
+      {:status (or (-> e ex-data :status-code)
+                   500)
+       :body   {:message (or (ex-message e)
+                             (tru "There was an error deleting the table"))}})))
+
+(api/defendpoint ^:multipart DELETE "/:id"
+  "Delete the given table from the database."
```

Comment:
This might need more clarification that it'll only deletes the csv uploaded tables.

Reply:
Sure, I'll update the comment to mention that it currently is limited to uploaded tables.